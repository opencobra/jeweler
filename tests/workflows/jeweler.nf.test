nextflow_workflow {

    name 'Test Workflow JEWELER'
    script 'workflows/jeweler.nf'
    workflow 'JEWELER'
    tag 'workflows'
    tag 'jeweler'

    test('Should generate a report snapshot without failure') {
        when {
            params {
                outdir = "$outputDir"
            }
            workflow {
                """
                input[0] = Channel.of(
                    [[id: 'e_coli_core'], file('https://raw.githubusercontent.com/opencobra/jeweler/testdata/data/model/e_coli_core.xml.gz')]
                )
                input[1] = true
                input[2] = false
                """
            }
        }

        then {
            assert workflow.success

            assert workflow.out.report_snapshot.size() == 1
            assert workflow.out.report_raw.size() == 0
            assert workflow.out.validation.size() == 1

            assert snapshot(workflow.out.versions).match('snapshot-versions')
        }
    }

    test('Should generate a raw JSON report without failure') {
        when {
            params {
                outdir = "$outputDir"
            }
            workflow {
                """
                input[0] = Channel.of(
                    [[id: 'e_coli_core'], file('https://raw.githubusercontent.com/opencobra/jeweler/testdata/data/model/e_coli_core.xml.gz')]
                )
                input[1] = false
                input[2] = true
                """
            }
        }

        then {
            assert workflow.success

            assert workflow.out.report_snapshot.size() == 0
            assert workflow.out.report_raw.size() == 1
            assert workflow.out.validation.size() == 1

            assert snapshot(workflow.out.versions).match('raw-versions')
        }
    }

    test('Should generate both reports without failure') {
        when {
            params {
                outdir = "$outputDir"
            }
            workflow {
                """
                input[0] = Channel.of(
                    [[id: 'e_coli_core'], file('https://raw.githubusercontent.com/opencobra/jeweler/testdata/data/model/e_coli_core.xml.gz')]
                )
                input[1] = true
                input[2] = true
                """
            }
        }

        then {
            assert workflow.success

            assert workflow.out.report_snapshot.size() == 1
            assert workflow.out.report_raw.size() == 1
            assert workflow.out.validation.size() == 1

            assert snapshot(workflow.out.versions).match('snapshot-raw-versions')
        }
    }
}
