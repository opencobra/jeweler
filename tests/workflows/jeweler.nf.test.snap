{
    "snapshot-versions": {
        "content": [
            [
                "versions.yml:md5,8415a0b802a382e43ab1e162929f37a3",
                "versions.yml:md5,c29dae7fe915e975714c56b92f8885ff"
            ]
        ],
        "timestamp": "2023-11-28T22:16:07.263034958"
    },
    "snapshot-raw-versions": {
        "content": [
            [
                "versions.yml:md5,1b2fe0ed8b75f06aa2d41d37dfb00f17",
                "versions.yml:md5,8415a0b802a382e43ab1e162929f37a3",
                "versions.yml:md5,c29dae7fe915e975714c56b92f8885ff"
            ]
        ],
        "timestamp": "2023-11-28T22:16:39.008683712"
    },
    "raw-versions": {
        "content": [
            [
                "versions.yml:md5,1b2fe0ed8b75f06aa2d41d37dfb00f17",
                "versions.yml:md5,8415a0b802a382e43ab1e162929f37a3"
            ]
        ],
        "timestamp": "2023-11-28T22:16:22.414753979"
    }
}